projects/packages/publicize/src/class-publicize.php

```diff
@@ -600,6 +600,7 @@ public function test_connection( $service_name, $connection ) {
 
 		$xml_response            = $xml->getResponse();
 		$connection_test_message = $xml_response['faultString'];
+		$connection_error_code   = $xml_response['faultCode'];
```

Comment:
Oh no, that's not great! You're right, it comes back as 0 I think, as the IXRError class tries to parse the error code as an int, but it's a string, so it's converted to 0.

Sounds like we should make it `'pub_conn_test_failed'` again if the `faultCode` is `empty`.

How serious is this? Should we be reverting it?

@gmjuhasz Could you have a look at fixing it?

Reply:
The errors weren't enough to have to revert the 12.2-a.5 rollout to Atomic, so we should be ok on that front. I have no idea what effect if any this might have for users.

It would be good to have a fix in before Monday, to make it into 12.2-a.7. Or maybe it can be entirely fixed on the wpcom side where the error response is generated?

Reply:
Sorry, my Github notifications got messed up! I can take a look at this tomorrow morning. It's definitely the issue I described in the description too, I'll try to fix it on WPCOM